app/Http/Controllers/Table/RoutesTablesController.php

```diff
@@ -0,0 +1,188 @@
+use App\Models\Port;
+use App\Models\Device;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Http\Request;
+use LibreNMS\Util\IP;
+use LibreNMS\Util\Rewrite;
+use LibreNMS\Util\Url;
+
+class RoutesTablesController extends TableController
+{
+    protected $ipCache = [];
+
+    protected function rules()
+    {
+        return [
+            'device_id' => 'nullable|integer',
+            'searchby' => 'in:inetCidrRouteNextHop,inetCidrRouteDest,inetCidrRouteProto,inetCidrRouteType',
+            ];
+    }
+
+    protected function filterFields($request)
+    {
+        return [
+            'inetCidrRoute.device_id' => 'device_id',
+            ];
+    }
+
+    /**
+     * Defines the base query for this resource
+     *
+     * @param \Illuminate\Http\Request $request
+     * @return \Illuminate\Database\Eloquent\Builder|\Illuminate\Database\Query\Builder
+     */
+    protected function baseQuery($request)
+    {
+        return InetCidrRoute::hasAccess($request->user())
+            ->with(['device'])
+            ->select('inetCidrRoute.*');
+    }
+
+    /**
+     * @param string $search
+     * @param Builder $query
+     * @param array $fields
+     * @return Builder|\Illuminate\Database\Query\Builder
+     */
+    protected function search($search, $query, $fields = [])
+    {
+        if ($search = trim(\Request::get('searchPhrase'))) {
+            $searchLike = '%' . $search . '%';
+            return $query->where('inetCidrRoute.inetCidrRouteNextHop', 'like', $searchLike)
+                ->orWhere('inetCidrRoute.inetCidrRouteDest', 'like', $searchLike);
+        }
+
+        return $query;
+    }
+
+    /**
+     * @param Request $request
+     * @param Builder $query
+     * @return Builder
+     */
+    public function sort($request, $query)
+    {
+        $sort = $request->get('sort');
+
+        if (isset($sort['inetCidrRouteIfIndex'])) {
+            $query->join('ports', function ($join) {
+                    $join->on('inetCidrRoute.inetCidrRouteIfIndex', 'ports.ifIndex')
+                    ->on('inetCidrRoute.device_id', '=', 'ports.device_id');
+            })
+            ->orderBy('ports.ifDescr', $sort['inetCidrRouteIfIndex']);
+        }
+
+        if (isset($sort['inetCidrRouteProto'])) {
+            $query->orderBy('inetCidrRouteProto', $sort['inetCidrRouteProto']);
+        }
+        if (isset($sort['inetCidrRouteType'])) {
+            $query->orderBy('inetCidrRouteType', $sort['inetCidrRouteType']);
+        }
+
+        if (isset($sort['updated_at'])) {
+            $query->orderBy('updated_at', $sort['updated_at']);
+        }
+
+        if (isset($sort['created_at'])) {
+            $query->orderBy('created_at', $sort['created_at']);
+        }
+
+        if (isset($sort['context_name'])) {
+            $query->orderBy('context_name', $sort['context_name']);
+        }
+
+        return $query;
+    }
+
+    public function formatItem($route_entry)
+    {
+        $translateProto[1] = '1-other';
```

Comment:
You should define these arrays outside of the function (otherwise they get defined each time it is called)
and better to define like this:
```php
$translateProto = [
    1 => '1-other',
    ...
];
```

Reply:
I moved it as static in the class.